### ISSUE TITLE
Doom Relaod does not reload emacs

### ISSUE THREAD

**Issue by Amrit-codes** [doomemacs/doomemacs#7636]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

space h r r should reload the config

### Current behavior

The changes made in the config file only take place after I kill the emacs client and reopen it

### Steps to reproduce

1. Start emacsclient
2. Add another language or package to the init.el or packages.el file
3. On pressing space h r r the config is reloaded
4. Even though the config is reloaded, the new packages or language do not work until the emacsclient is killed and restarted 

### System Information

https://pastebin.com/a2ULFPY7

**Comment by RBckmnn** [comment]
I'm pretty sure, you have to run `doom sync` manually after making changes to `packages.el` or `init.el`.

See

https://github.com/doomemacs/doomemacs/blob/master/docs/faq.org#know-when-to-run--doom-sync

**Comment by Amrit-codes** [comment]
I tried running doom sync and doom reload after making changes, the auto completion was detecting the commands but I couldn't run neotree command till I killed emacs daemon and restarted it.

**Comment by 45mm** [comment]
I've had the same issue since I started using Doom (~4 months ago). I also use emacsclient.

**Comment by yaaama** [comment]
> I've had the same issue since I started using Doom (~4 months ago). I also use emacsclient.


> I tried running doom sync and doom reload after making changes, the auto completion was detecting the commands but I couldn't run neotree command till I killed emacs daemon and restarted it.


I use emacsclient myself and I just use "SPC q K" to kill the client and server, I have a helper script to manage the emacs daemon that lets me launch it up and a frame again (i've got it on gh if it is useful for anyone).

I'm unsure if there is anything we can do apart from a full server restart to be honest

**Comment by davidhaley** [comment]
I've experienced the same thing since I started using Doom, and I kind of got into the habit of just restarting the Emacs daemon (to remove the guesswork of knowing whether or not a change took effect).

`killall emacs && emacs --daemon`

Also, I'll leave this helper-script here in case it helps anyone else. I created an i3 binding to call it.

If a Doom Emacs client is running when I call it, it will restart the daemon and then open a Doom client. If a Doom client was not running when I call it, it will just restart the daemon.

`restart-emacs-daemon.sh`:

```
#!/bin/bash

# Path to a temporary file to store if a Doom emacsclient was running
doom_client_state="/tmp/doom_client_running"

# Initially assume no Doom client is running
echo "false" > "$doom_client_state"

# Use xdotool to search for any window with Doom Emacs properties
if xdotool search --name "Doom Emacs" > /dev/null 2>&1; then
  echo "true" > "$doom_client_state"
fi

# Check if Emacs daemon is running and kill it if it is
if pgrep -x "emacs" > /dev/null; then
  pkill -x "emacs"
  # Wait for the Emacs daemon to fully terminate
  while pgrep -x "emacs" > /dev/null; do sleep 1; done
fi

# Restart the Emacs daemon
emacs --daemon

# If a Doom Emacs client was previously running, open it again
if [ "$(cat "$doom_client_state")" = "true" ]; then
  emacsclient -cn &
fi

# Clean up
rm "$doom_client_state"
```

And then in `~/.config/i3/config`:
```
bindsym $mod+Shift+d exec "/<path-to>/restart-emacs-daemon.sh"
```

**Comment by 45mm** [comment]
> I use emacsclient myself and I just use "SPC q K" to kill the client and server, I have a helper script to manage the emacs daemon that lets me launch it up and a frame again (i've got it on gh if it is useful for anyone).

Yeah, this is what I do as well. If you use a systemd unit as suggested [in the Emacs Wiki](https://www.emacswiki.org/emacs/EmacsAsDaemon#h5o-1) it's super easy - `Restart=on-failure` means you can just kill Emacs and the server starts right up again.

I guess it would be nice to not have to wait  through ~15s startup every time I want to test if my modified config loads without error, but I'm sure there's higher-priority work to be done right now . The docstring for the command does say it's experimental anyway.

**Comment by thigsu** [comment]
I'm also having the same problem since the first day of installation. I thought it was a problem with my system ( I'm using Windows ) so I reinstalled everything a couple of times and even did a clean install of Windows to see if it worked, but still not getting the "Config successfully reloaded!" message.

**Comment by hlissner** [comment]
As of c0c52f0 this should be resolved. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
